### PR TITLE
[YUNIKORN-396] Add the checklist in our release-repo documentation

### PR DESCRIPTION
**Pull request by adamantal**

### CODE REVIEW

File: docs/release-procedure.md [hunk 1]
```diff
@@ -106,12 +106,12 @@ Once the voting is passed, move the release artefacts to https://dist.apache.org
 Once moved to this space, the content will be automatically synced to https://downloads.apache.org/incubator/yunikorn/ which can be used as the final location for release files.
 Read more for [location of files on main server](https://infra.apache.org/mirrors#location).
 
-Publish an announcement blog to https://blogs.apache.org/yunikorn/, update the web-site with corresponding
-release notes, download links.
+## Release Docker images
+
+Push the latest docker images to docker hub.
```
Comment (yangwwei) [inline docs/release-procedure.md:111]: I think we should mention the tagged version here. Make sure the docker image is built on the specific SHA.

File: docs/release-procedure.md [hunk 2]
```diff
@@ -121,3 +121,13 @@ Fore more information please check [Helm documentation](https://helm.sh/docs/top
 - upload the packaged chart to the release in this repository
 - update the [index.yaml](https://github.com/apache/incubator-yunikorn-release/blob/gh-pages/index.yaml) file in the gh-pages branch with the new release
 
+## Update the website
+
+- Create a new version of the [YuniKorn website](https://github.com/apache/incubator-yunikorn-site/tree/master).
```
Comment (yangwwei) [inline docs/release-procedure.md:126]: I think this is better to say "Create a new documentation version on YuniKorn website based on the latest content in [docs](https://github.com/apache/incubator-yunikorn-site/tree/master/docs) directory."
Pls link to https://github.com/apache/incubator-yunikorn-site/tree/master#release-a-new-version

File: docs/release-procedure.md [hunk 3]
```diff
@@ -121,3 +121,13 @@ Fore more information please check [Helm documentation](https://helm.sh/docs/top
...
+- Update the [download page](http://yunikorn.apache.org/community/download) of the website.
+- Publish release announcement in the [apache Blog](https://blogs.apache.org/yunikorn/) and the the [YuniKorn website](http://yunikorn.apache.org).
```
Comment (yangwwei) [inline docs/release-procedure.md:128]: there is an extra "the"
and we should mention the release notes should be under versioned doc, like: http://yunikorn.apache.org/docs/get_started/release_notes

**Review by yangwwei — CHANGES_REQUESTED** [review]

File: docs/release-procedure.md (hunk 1 above)
Comment (adamantal) [reply docs/release-procedure.md]: Added.

File: docs/release-procedure.md (hunk 2 above)
Comment (adamantal) [reply docs/release-procedure.md]: Added the sentence and the link

File: docs/release-procedure.md (hunk 3 above)
Comment (adamantal) [reply docs/release-procedure.md]: Ah indeed. Added the extra mention as well.

**Comment by adamantal** [comment]
Thanks for the review @yangwwei . Would you please review it if you have some time?

**Review by yangwwei — APPROVED** [review]

**Pull request merged**